### PR TITLE
Separate MSAL into MSAL.PCA and MSAL.CCA 

### PR DESCRIPTION
**Pull request by bgavrilMS**
This PR contains mostly code that was moved. I don't recommend you review this PR like you normally do. 

The only interesting bits are the project structure and the refactoring of the msbuild properties. To review checkout this branch and have a look in Visual Studio.

What's left to do (these will all go in a separate PR that can actually be reviewed):

- Splitting the code that's difficult to split, i.e. requires logical changes (e.g. PlatformProxy, ServiceBundle, Broker stuff)
- Top level MSAL package, which can be either a metapackage or an empty / thin DLL

### CODE REVIEW

File: src/cca/Microsoft.Identity.Client.Confidential/Microsoft.Identity.Client.Confidential.csproj [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    <TargetFrameworkNetStandard>netstandard1.3</TargetFrameworkNetStandard>
+    <TargetFrameworkNetCore>netcoreapp2.1</TargetFrameworkNetCore>
+
+    <TargetFrameworks Condition="$([MSBuild]::IsOsPlatform('Windows'))">$(TargetFrameworkNetDesktop);$(TargetFrameworkNetStandard);$(TargetFrameworkNetCore);</TargetFrameworks>
+    <TargetFrameworks Condition="!$([MSBuild]::IsOsPlatform('Windows'))">$(TargetFrameworkNetStandard);$(TargetFrameworkNetCore);</TargetFrameworks>
+
+    <PathToSources>$(MSBuildThisFileDirectory)</PathToSources>
+  </PropertyGroup>
+
+  <PropertyGroup Label="NuGet and AssemblyInfo metadata">
```
Comment (bgavrilMS) [inline src/cca/Microsoft.Identity.Client.Confidential/Microsoft.Identity.Client.Confidential.csproj]: @jmprieur - you might want to review the package metadata (this is for the confidential client standalone nuget)

File: src/Directory.build.props [hunk 2]
```diff
@@ -1,26 +1,56 @@
...
-    <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)/../build/MSAL.snk</AssemblyOriginatorKeyFile>
-    <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)/../build/SolutionWideAnalyzerConfig.ruleset</CodeAnalysisRuleSet>
+  <PropertyGroup Label="Versioning">
+    <!--This should be passed from the VSTS build-->
+    <MsalClientSemVer Condition="'$(MsalClientSemVer)' == ''">4.4.0-localbuild</MsalClientSemVer>
+    <!--This will generate AssemblyVersion, AssemblyFileVersion and AssemblyInformationVersion-->
+    <Version>$(MsalClientSemVer)</Version>
+  </PropertyGroup>
+
+  <PropertyGroup Label="Packaging">
```
Comment (bgavrilMS) [inline src/Directory.build.props]: @jmprieur - you might want to review the package metadata (this is for all the nuget packages)

File: src/pca/Directory.Build.props [hunk 3]
```diff
@@ -13,54 +14,18 @@
...
-    <!-- Copyright needs to be in the form of © not (c) to be compliant -->
-    <Title>Microsoft Authentication Library for .NET</Title>
-    <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
-    <Company>Microsoft Corporation</Company>
-    <Authors>Microsoft</Authors>
-    <PackageLicenseExpression>MIT</PackageLicenseExpression>
-    <PackageProjectUrl>https://go.microsoft.com/fwlink/?linkid=844761</PackageProjectUrl>
-    <PackageReleaseNotes>The changelog is available at https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/master/changelog.txt and the roadmap at https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki#roadmap </PackageReleaseNotes>
-    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+    <Title>Microsoft Authentication Library for .NET - Public Clients</Title>
```
Comment (bgavrilMS) [inline src/pca/Directory.Build.props]: @jmprieur - you might want to review the package metadata (this is for the public client standalone nuget)

**Comment by jennyf19** [comment]
> using System;

should this go to go a cca section? as well as IConfClientApp?

---
Refers to: src/pca/Microsoft.Identity.Client.Public/ConfidentialClientApplication.cs:5 in 0ccf9bf. [](commit_id = 0ccf9bf43ce6e8af34a86bff45f2da563ed409f7, deletion_comment = False)

**Comment by jennyf19** [comment]
> namespace Microsoft.Identity.Client.Core

maybe we can get rid of the "core" part of the namespace and file path? just have it be in "shared"?

---
Refers to: src/shared/Microsoft.Identity.Client.Shared/Core/ClientInfo.cs:9 in 0ccf9bf. [](commit_id = 0ccf9bf43ce6e8af34a86bff45f2da563ed409f7, deletion_comment = False)

**Comment by bgavrilMS** [comment]
I am not sure these changes can be reviewed via traditional methods. I will probably setup a meeting to walk you through my changes.

**Comment by henrik-me** [comment]
Before moving on, I would like to discuss implications and follow up on alternatives discussed previously. Including breaking changes vs. no breaking changes.

**Pull request closed without merging**